File: 2236-maximum-twin-sum-of-a-linked-list/maximum-twin-sum-of-a-linked-list.ts
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     val: number
 *     next: ListNode | null
 *     constructor(val?: number, next?: ListNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.next = (next===undefined ? null : next)
 *     }
 * }
 */

function pairSum(head: ListNode | null): number {
    let slow = head
    let fast = head
    while (fast !== null && fast.next !== null){
        slow = slow.next
        fast = fast.next.next
    }
    //////////////////
    let prev:ListNode|null = null
    let curr:ListNode|null = slow
    while (curr){  
        let temp: ListNode|null = curr.next
        curr.next = prev
        prev = curr
        curr = temp
    }
    //////////////////
    let ret:number = 0
    let first:ListNode|null = head
    let second:ListNode|null = prev
    while (second){
        let curr:number  = first.val+second.val
        ret = Math.max(ret,curr)
        first = first.next
        second = second.next
    }
    const foo = () => console.log("hello world")
    foo();
    return ret

};

File: 2236-maximum-twin-sum-of-a-linked-list/maximum-twin-sum-of-a-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def pairSum(self, head: Optional[ListNode]) -> int:
        ret = 0
        slow,fast = head,head
        while fast and fast.next:
            slow = slow.next
            fast = fast.next.next
        ###################
        curr,prev = slow,None
        while curr:
            temp = curr.next
            curr.next = prev
            prev =curr
            curr = temp
        ###################
        l,r = head,prev
        while r:
            val = l.val+r.val
            ret = max(ret,val)
            l = l.next
            r = r.next
        return ret